scripting/tf2attribute_support.sp

```diff
@@ -1152,7 +1152,7 @@ MRESReturn OnPlayerKilledPre(int client, Handle hParams) {
 			ObjectValueType_Ehandle);
 	
 	g_bShouldTurnToIce[client] = IsValidEntity(weapon)
-			&& TF2Attrib_HookValueInt(0, "set_turn_to_ice", weapon);
+			&& HasEntProp(weapon, Prop_Send, "m_AttributeList") && TF2Attrib_HookValueInt(0, "set_turn_to_ice", weapon);
```

Comment:
`HasEntProp` was historically not performant; I think it'd be better to check with `!weapon && (TF2Util_IsEntityWeapon(weapon) || TF2Util_IsEntityWearable(weapon))` (the wearable check is for shields, and `!weapon` avoids calling the tf2utils natives for world).

Reply:
I changed it to this.